### ISSUE TITLE
 salt master performance degradation when there's a symlink to network filer containing 1000s of files

### ISSUE THREAD

**Issue by COcjeske** [saltstack/salt#27576]
After adding a symbolic link to the master's file server root directory, a significant performance degradation was experienced. When attempting to perform test.ping or state.highstate we experienced "failed to authenticate" errors, time-outs, inability to access salt directories from minions (even though they existed), etc.  After removing the symlink performance returned to normal. 

number of minions: ~500

excerpt from /etc/salt/master:

``` yaml
   file_roots:
     base:
       - /srv/salt/base
```

`/srv/salt/base/<dir>  ->  <mount point to a network filer containing ~3000 files>`

``` yaml
Version info:
   Salt: 2014.1.13
   Python: 2.7.3 (default, Feb 27 2014, 19:58:35)
   Jinja2: 2.6
  M2Crypto: 0.21.1
  msgpack-python: 0.1.10
  msgpack-pure: Not Installed
  pycrypto: 2.6.1
  PyYAML: 3.10
  PyZMQ: 13.1.0
  ZMQ: 3.2.3
```

**Comment by jtand** [comment]
@COcjeske Can you reproduce this on a newer version of salt, like 2015.5.5 or 2015.8.0? Thanks!

**Comment by ogabrielsantos** [comment]
I've spent the day debugging and trying to solve minions who have stopped responding abruptly until I find this issue. I've created symlinks in order to workaround https://github.com/saltstack-formulas/bind-formula/issues/37 's hardcoded settings:

````
file_roots:
  dev:
    - /srv/salt/prod/states/files/var/
    - [...]

/srv/salt/prod/states/files/var/
├── bind -> .
├── A LOT OF ZONE FILES.db
└── zones -> .
````

So, salt://bind/zones/A ZONE FILE.db  should return a zone file inside the first file_roots.

When I read this issue imediatly removed hacky `file_roots` and symlinks which solved the problem. All minions returned back! \o/

Maybe a lot of  `master not responding` issue is related to this?
